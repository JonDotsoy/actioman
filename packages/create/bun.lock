{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "create-actioman",
      "dependencies": {
        "@jondotsoy/flags": "^2.3.0",
      },
      "devDependencies": {
        "@types/bun": "latest",
      },
      "peerDependencies": {
        "typescript": "^5.0.0",
      },
    },
  },
  "packages": {
    "@jondotsoy/console-draw": ["@jondotsoy/console-draw@0.1.2", "", { "dependencies": { "@jondotsoy/style-text": "^0.9.1" }, "peerDependencies": { "prettier": "^3.3.3", "typescript": "^5.0.0" } }, "sha512-BBZGLd6RzjKxKcgD2OwD7v6GeKlf53Ysk59H+7+HvyGLcJZ7AJk8rOCq/yv813KbWV2teTCEBQXnnP5hZe/2CA=="],

    "@jondotsoy/flags": ["@jondotsoy/flags@2.3.0", "", { "dependencies": { "@jondotsoy/console-draw": "^0.1.2" } }, "sha512-vUMe4NdXKv8RxMXrKCJLsHBt3XQ8/YV4Fyr1I43vSLe5v0M8sdY3wJl1qdTgwRBZES2zMLoenxOUSNGE9r/5NA=="],

    "@jondotsoy/style-text": ["@jondotsoy/style-text@0.9.2", "", { "peerDependencies": { "typescript": "^5.0.0" } }, "sha512-6fonOzrANwrw+UIM4fivtDZLvuCN00yH0lCEYi14ln4j8fSHZokb6TLrqOu38GCqqQnZ2LuzJTQBWgv7jN6iVA=="],

    "@types/bun": ["@types/bun@1.2.2", "", { "dependencies": { "bun-types": "1.2.2" } }, "sha512-tr74gdku+AEDN5ergNiBnplr7hpDp3V1h7fqI2GcR/rsUaM39jpSeKH0TFibRvU0KwniRx5POgaYnaXbk0hU+w=="],

    "@types/node": ["@types/node@22.13.1", "", { "dependencies": { "undici-types": "~6.20.0" } }, "sha512-jK8uzQlrvXqEU91UxiK5J7pKHyzgnI1Qnl0QDHIgVGuolJhRb9EEl28Cj9b3rGR8B2lhFCtvIm5os8lFnO/1Ew=="],

    "@types/ws": ["@types/ws@8.5.14", "", { "dependencies": { "@types/node": "*" } }, "sha512-bd/YFLW+URhBzMXurx7lWByOu+xzU9+kb3RboOteXYDfW+tr+JZa99OyNmPINEGB/ahzKrEuc8rcv4gnpJmxTw=="],

    "bun-types": ["bun-types@1.2.2", "", { "dependencies": { "@types/node": "*", "@types/ws": "~8.5.10" } }, "sha512-RCbMH5elr9gjgDGDhkTTugA21XtJAy/9jkKe/G3WR2q17VPGhcquf9Sir6uay9iW+7P/BV0CAHA1XlHXMAVKHg=="],

    "prettier": ["prettier@3.4.2", "", { "bin": { "prettier": "bin/prettier.cjs" } }, "sha512-e9MewbtFo+Fevyuxn/4rrcDAaq0IYxPGLvObpQjiZBMAzB9IGmzlnG9RZy3FFas+eBMu2vA0CszMeduow5dIuQ=="],

    "typescript": ["typescript@5.7.3", "", { "bin": { "tsc": "bin/tsc", "tsserver": "bin/tsserver" } }, "sha512-84MVSjMEHP+FQRPy3pX9sTVV/INIex71s9TL2Gm5FG/WG1SqXeKyZ0k7/blY/4FdOzI12CBy1vGc4og/eus0fw=="],

    "undici-types": ["undici-types@6.20.0", "", {}, "sha512-Ny6QZ2Nju20vw1SRHe3d9jVu6gJ+4e3+MMpqu7pqE5HT6WsTSlce++GQmK5UXS8mzV8DSYHrQH+Xrf2jVcuKNg=="],
  }
}
