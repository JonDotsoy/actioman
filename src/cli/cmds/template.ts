import {
  flag,
  flags,
  isBooleanAt,
  makeHelpMessage,
  rule,
  type Rule,
} from "@jondotsoy/flags";

export const template = async (args: string[]) => {
  type Options = { help: boolean };
  const rules: Rule<Options>[] = [
    rule(flag("-h", "--help"), isBooleanAt("help"), {
      description: "Show help",
    }),
  ];
  const options = flags(args, {}, rules);

  if (options.help) return makeHelpMessage("actioman", rules);
};
