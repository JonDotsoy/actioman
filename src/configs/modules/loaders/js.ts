export default async (sourceLocation: string) => {
  const { default: def } = await import(sourceLocation);
  return def;
};
