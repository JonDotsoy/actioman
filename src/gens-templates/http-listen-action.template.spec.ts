import { describe, expect, it } from "bun:test";
import { httpListenActionTemplate } from "./http-listen-action.template.js";

describe("httpListenActionTemplate", () => {
  it("generates bootstrap script", () => {
    expect(
      httpListenActionTemplate({
        target: "file://app/scripts/serve.js",
        workspaceLocation: "file://app/",
        actionFileLocation: "file://mys-actions/actios-1/actions.js",
        modules: {
          httpListenerModuleLocation: new URL(
            "file://app/mode_modules/actioman/lib/esm/http-router/http-listener.js",
          ).toString(),
          configsFactoryModuleLocation: new URL(
            "file://app/mode_modules/actioman/lib/esm/configs/configs-factory.js",
          ).toString(),
          configsModuleLocation: new URL(
            "file://app/mode_modules/actioman/lib/esm/configs/configs.js",
          ).toString(),
        },
      }),
    ).toMatchSnapshot();
  });
});
