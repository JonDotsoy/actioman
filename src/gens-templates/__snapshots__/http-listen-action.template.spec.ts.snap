// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`httpListenActionTemplate generates bootstrap script 1`] = `
"// @ts-nocheck
import { HTTPLister } from './../mode_modules/actioman/lib/esm/http-router/http-listener.js';
import { factory } from './../undefined';
const PORT = process.env.PORT;
const HOST = process.env.HOST;
const bootstrap = async () => {
    const configs = Configs.fromModule(await factory.findOn(new URL('./../undefined/', import.meta.url).toString(), 'configs'));
    const httpLister = await HTTPLister.fromModule(await import('./../actios-1/actions.js'), configs);
    const url = await httpLister.listen(PORT, HOST);
    console.log(\`Server running at \${ url.toString() }\`);
};
bootstrap().catch(console.error);"
`;
